api/data/network-fees.js

```diff
@@ -54,8 +59,31 @@ const pollForNewFees = async () => {
   }, FEES_POLLING_INTERVAL)
 }
 
+const getNetworkGasPrices = async (networkId) => {
+  const networkGasPrices = await db.getNetworkGasPrices()
+  networkGasPrices.forEach(networkGasPrice => {
+    if (!networkGasPricesDictionary[networkGasPrice.id]) networkGasPricesDictionary[networkGasPrice.id] = []
+    networkGasPricesDictionary[networkGasPrice.id].push({ denom: networkGasPrice.denom,
+      gasPrice: networkGasPrice.gasPrice
+    })
+  })
+  // update emoneyGasPrices
+  networkGasPricesDictionary['emoney-mainnet'] = emoneyGasPrices
+  return networkGasPricesDictionary[networkId] || []
+}
+
+const getNetworkGasEstimates = async (networkId) => {
+  const networkGasEstimates = await db.getNetworkGasEstimates()
+  networkGasEstimates.forEach(networkGasEstimate => {
+    if (!networkGasEstimatesDictionary[networkGasEstimate.id]) networkGasEstimatesDictionary[networkGasEstimate.id] = {}
+    networkGasEstimatesDictionary[networkGasEstimate.id][networkGasEstimate.transactionType] = networkGasEstimate.gasEstimate
+  })
+  return networkGasEstimatesDictionary[networkId]
+}
+
 // run on API launch
 pollForNewFees()
+getNetworkGasEstimates()
```

Comment:
this doesn't return anything. what is the purpose here?

Reply:
the purpose is to have network gas estimates already loaded in `networkGasEstimatesDictionary`. Otherwise it would be undefined in `getNetworkTransactionGasEstimates`, needed to calculate transaction fees

Reply:
We run it on API launch